File: src/java/leetcode/editor/en/P1_TwoSum.java
package java.leetcode.editor.en;
import java.util.*;
import java.util.function.*;
import java.leetcode.editor.en.definition.*;

public class P1_TwoSum {
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int y = target - nums[i];
            if (m.containsKey(y)) return new int[] {m.get(y), i};
            m.put(nums[i], i);
        }
        throw new IllegalArgumentException();
    }
}
//leetcode submit region end(Prohibit modification and deletion)
}

File: src/kt/leetcode/editor/en/P1_TwoSum.kt
package kt.leetcode.editor.en;
import java.util.*;
import java.util.function.*;
import kt.leetcode.editor.en.definition.*;
import java.lang.IllegalArgumentException
import java.leetcode.editor.en.definition.ListNode
import kotlin.collections.HashMap

public class P1_TwoSum {
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()
        nums.forEachIndexed { index, num ->
            map[target-num]?.let { return intArrayOf(it, index) }
            map[num] = index
        }
        throw IllegalArgumentException()
    }
}
//leetcode submit region end(Prohibit modification and deletion)
}
